python/chemiscope/structures/_stk.py

```diff
@@ -18,7 +18,7 @@ def _stk_valid_structures(
     if HAVE_STK and isinstance(frames, Molecule):
         # deal with the user passing a single frame
         return [frames], True
-    elif HAVE_STK and isinstance(frames[0], Molecule):
+    elif HAVE_STK and hasattr(frames, "__iter__") and isinstance(frames[0], Molecule):
```

Comment:
Why this check? The following `isinstance(frames[0], Molecule)` should already return early when using non stk data.

Reply:
Mainly because `Universe` is not an iterable. Without this check, a `TypeError` will be thrown here if we passes an `Universe`.

Reply:
Is it also not indexable? i.e. `universe[0]` fails? I would rather check for this by doing 
```suggestion
    elif HAVE_STK and isinstance(frames, list) and isinstance(frames[0], Molecule):
```

Reply:
Yes not indexable